src/main/java/com/example/home_work_8/gameplay/Treadmill.java

```diff
@@ -0,0 +1,44 @@
+package com.example.home_work_8.gameplay;
+
+public class Treadmill implements Obstacle {
+
+    private int distance;
+    final String name;
+
+    public Treadmill(int distance) {
+        this.name = "Treadmill";
+        this.distance = distance;
+    }
+
+    @Override
+    public int getDistance() {
+        return distance;
+    }
+
+    @Override
+    public int getHight() {
+        return 0;
+    }
+
+    public void setDistance(int distance) {
+        this.distance = distance;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public void overcome() {
```

Comment:
this method should accept the parameter `Participant participant` and you should check whether that participant can overcome the distance, if yes then print the appropriate message, if not print another message that the participant failed to overcome the obstacle